Makefile.am

```diff
@@ -636,8 +636,10 @@ define make_parent_dir
 endef
 
 define make_tss_user_and_group
-    (id -g tss 2>/dev/null || groupadd -r tss) && \
-    (id -u tss 2>/dev/null || useradd -r -g tss tss)
+    (if ! type -p groupadd > /dev/null; then (id -g tss 2>/dev/null || addgroup -S tss); fi) && \
+    (if ! type -p useradd > /dev/null; then (id -u tss 2>/dev/null || adduser -S -G tss tss); fi) && \
+    (if type -p groupadd > /dev/null; then (id -g tss 2>/dev/null || groupadd -r tss); fi) && \
+    (if type -p useradd > /dev/null; then (id -u tss 2>/dev/null || useradd -r -g tss tss); fi)
```

Comment:
Read my mind.

Reply:
Actually, I misread this. I would do this:
```
# Use the higher level friendly tools over non-friendly useradd/groupadd
if addgroup_defined:
  addgroup
else:
  addgroup linux-way || addgroup busybox-way

if adduser_defined:
  adduser 
else:
  adduser linux-way || adduser busybox-way
```

Extra points to keep track of stderr and only display if both commands fail.